modular_zubbers/code/modules/antagonists/bloodsucker/bloodsuckers/procs.dm

```diff
@@ -165,6 +165,26 @@
 	//returnString += "\n"  Don't need spacers. Using . += "" in examine.dm does this on its own.
 	return returnIcon + returnString
 
+// Blood level gain is used to give Bloodsuckers more levels if they are being agressive and drinking from real, sentient people.
+// The maximum blood that counts towards this
+/datum/antagonist/bloodsucker/proc/blood_level_gain()
+	var/level_cost = get_level_cost()
+	if(blood_level_gain >= level_cost && bloodsucker_blood_volume >= level_cost) // Checks if we have drunk enough blood from the living to allow us to gain a level up as well as checking if we have enough blood to actually use on the level up
+		switch(tgui_alert(owner.current, "You have drunk enough blood from the living to thicken your blood, this will cost you [level_cost] blood and give you another level",  "Thicken your blood?.", list("Yes", "No"))) //asks user if they want to spend their blood on a level
+			if("Yes")
+				bloodsucker_level_unspent++ // gives level
+				blood_level_gain -= level_cost // Subtracts the cost from the pool of drunk blood
+				AdjustBloodVolume(-level_cost) // Subtracts the cost from the bloodsucker's actual blood
+				blood_level_gain_amount += 1 // Increments the variable that makes future levels more expensive
+			if("No")
+				return
```

Comment:
This can just be an early return if it isn't "Yes" because switch can be x'ed out which would return neither yes or no, and there's no reason to specifically call return anyway.

Reply:
Should be fixed.